[Linux] Update to 6.1.23 makes application instant crash

### Current behavior 🐛

Upgrading to 6.1.23 makes application crash on unmanaged code as soon as it starts rendering (on `host.Run()`, everything previous to that executes).

### Expected behavior 🎯

Application should work with no changes.

### How to reproduce it (as minimally and precisely as possible) 🔬

Change SDK version in `global.json` to 6.1.23.

### Workaround 🛠️

_No response_

### Renderer 🎨

- [x] Skia
- [ ] Native

### Affected platforms 📱💻🖥️

Desktop (X11)

### Uno.Sdk version (and other relevant versions) 📦

```json
{
"sdk":          {
"version":         "9.0.0",
"rollForward":     "latestMajor",
"allowPrerelease": true
},
"msbuild-sdks": {
"Uno.Sdk": "6.1.23"
}
}
```

### IDE version 🧑‍💻

Rider 2025.2 EAP8

### Anything else we need to know? 💬

GDB output:
```
claunia@serket ~/Development/ClauRepairDb/ClauRepairDb/bin/Debug/net9.0-desktop [main !] $ gdb --args ./ClauRepairDb
GNU gdb (GDB) 16.3
Copyright (C) 2024 Free Software Foundation, Inc.
License GPLv3+: GNU GPL version 3 or later <http://gnu.org/licenses/gpl.html>
This is free software: you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.
Type "show copying" and "show warranty" for details.
This GDB was configured as "x86_64-pc-linux-gnu".
Type "show configuration" for configuration details.
For bug reporting instructions, please see:
<https://www.gnu.org/software/gdb/bugs/>.
Find the GDB manual and other documentation resources online at:
<http://www.gnu.org/software/gdb/documentation/>.

For help, type "help".
Type "apropos word" to search for commands related to "word"...
Reading symbols from ./ClauRepairDb...

This GDB supports auto-downloading debuginfo from the following URLs:
<https://debuginfod.archlinux.org>
<https://debuginfod.cachyos.org>
Enable debuginfod for this session? (y or [n]) y
Debuginfod has been enabled.
To make this setting permanent, add 'set debuginfod enabled on' to .gdbinit.
(No debugging symbols found in ./ClauRepairDb)
(gdb) r
Starting program: /home/claunia/Development/ClauRepairDb/ClauRepairDb/bin/Debug/net9.0-desktop/ClauRepairDb
Function(s) ^std::(move|forward|as_const|(__)?addressof) will be skipped when stepping.
Function(s) ^std::(shared|unique)_ptr<.*>::(get|operator) will be skipped when stepping.
Function(s) ^std::(basic_string|vector|array|deque|(forward_)?list|(unordered_|flat_)?(multi)?(map|set)|span)<.*>::(c?r?(begin|end)|front|back|data|size|empty) will be skipped when stepping.
Function(s) ^std::(basic_string|vector|array|deque|span)<.*>::operator.] will be skipped when stepping.
[Thread debugging using libthread_db enabled]
Using host libthread_db library "/usr/lib/libthread_db.so.1".
[New Thread 0x7fff7800f6c0 (LWP 3981202)]
[New Thread 0x7fff7780e6c0 (LWP 3981203)]
[New Thread 0x7fff7700d6c0 (LWP 3981204)]
[New Thread 0x7fff7680c6c0 (LWP 3981205)]
[New Thread 0x7fff7600b6c0 (LWP 3981206)]
[New Thread 0x7fbecb54e6c0 (LWP 3981209)]
[New Thread 0x7fbec1fff6c0 (LWP 3981210)]
[New Thread 0x7fbecad4d6c0 (LWP 3981211)]
[New Thread 0x7fbec17fe6c0 (LWP 3981212)]
Downloading 717.41 K separate debug info for /usr/lib/libX11.so.6
Downloading 12.18 K separate debug info for /usr/lib/libXau.so.6
[New Thread 0x7fbeb3fff6c0 (LWP 3981213)]
[New Thread 0x7fbeb31ff6c0 (LWP 3981214)]
[New Thread 0x7fbeb1dff6c0 (LWP 3981215)]
[New Thread 0x7fbeb15fe6c0 (LWP 3981216)]
Downloading 992.98 K separate debug info for /usr/lib/libfreetype.so.6
Downloading 120.66 K separate debug info for /usr/lib/libexpat.so.1
Downloading 1.08 M separate debug info for /usr/lib/libz.so.1
Downloading 88.00 K separate debug info for /usr/lib/libbz2.so.1.0
Downloading 37.71 M separate debug info for /usr/lib/libharfbuzz.so.0
Downloading 60.83 K separate debug info for /usr/lib/libbrotlidec.so.1
Downloading 5.04 M separate debug info for /usr/lib/libglib-2.0.so.0
Downloading 11.77 K separate debug info for /usr/lib/libbrotlicommon.so.1
Downloading 489.66 K separate debug info for /usr/lib/libpcre2-8.so.0
Downloading 184.14 K separate debug info for /usr/lib/libGL.so.1
Downloading 328.37 K separate debug info for /usr/lib/libGLdispatch.so.0
Downloading 250.48 K separate debug info for /usr/lib/libGLX.so.0
Downloading 4.27 K separate debug info for /usr/lib/libX11-xcb.so.1
Downloading 323.16 K separate debug info for /usr/lib/libdbus-1.so.3
Downloading 4.56 M separate debug info for /usr/lib/libsystemd.so.0
Downloading 74.72 K separate debug info for /usr/lib/libdrm.so.2
Downloading 71.45 K separate debug info for /usr/lib/libXi.so.6
Downloading 36.08 K separate debug info for /usr/lib/libXcursor.so.1
Downloading 32.83 K separate debug info for /usr/lib/libXrender.so.1
Downloading 17.27 K separate debug info for /usr/lib/libXfixes.so.3
[New Thread 0x7fbea1f7f6c0 (LWP 3981223)]
[New Thread 0x7fbea177e6c0 (LWP 3981224)]
[New Thread 0x7fbea095c6c0 (LWP 3981225)]

Thread 17 "Uno X11 Renderi" received signal SIGSEGV, Segmentation fault.
[Switching to Thread 0x7fbea095c6c0 (LWP 3981225)]
0x00007fbea2f1d369 in glDisable () from /usr/lib/libGLdispatch.so.0
```

Linux distro: CachyOS rolling
X11 version: 24.1.8-1.1
Wayland version: 1.23.1-2.1
Kernel version: 6.15.4-4-cachyos
NVIDIA driver version: 575.64.03
.NET host: 10.0.100-preview.5.25277.114
.NET SDK: 9.0.302
.NET target platform: net9.0-desktop

Updating `global.json` with these contents also make it launch with `dotnet run -f net9.0-desktop`:

```json
{
  // To update the version of Uno please update the version of the Uno.Sdk here. See https://aka.platform.uno/upgrade-uno-packages for more information.
  "msbuild-sdks": {
    "Uno.Sdk": "6.0.146"
  },
  "sdk":{
    "allowPrerelease": true
  }
}
```

I was requested to test these, which also caused a SEGV:

- 6.2.0-dev.28
- 6.1.0-dev.268
- 6.1.0-dev.230

Then to find the source I also tested the following:

- 6.1.0-dev.4 -> `/home/lietu/src/cocreators/Example/Example/Platforms/Desktop/Program.cs(12,20): error CS0103: The name 'UnoPlatformHostBuilder' does not exist in the current context`
- 6.1.0-dev.30 -> `/home/lietu/src/cocreators/Example/Example/Platforms/Desktop/Program.cs(12,20): error CS0103: The name 'UnoPlatformHostBuilder' does not exist in the current context`
- 6.1.0-dev.100 -> OK
- 6.1.0-dev.165 ->
```
Unhandled exception. System.TypeInitializationException: The type initializer for '<Module>' threw an exception.
 ---> System.DllNotFoundException: Unable to load shared library 'USER32.dll' or one of its dependencies. In order to help diagnose loading problems, consider using a tool like strace. If you're using glibc, consider setting the LD_DEBUG environment variable: 
/home/lietu/src/cocreators/Example/Example/bin/Debug/net9.0-desktop/runtimes/linux-x64/native/USER32.dll.so: cannot open shared object file: No such file or directory
/usr/share/dotnet/shared/Microsoft.NETCore.App/9.0.5/USER32.dll.so: cannot open shared object file: No such file or directory
/home/lietu/src/cocreators/Example/Example/bin/Debug/net9.0-desktop/USER32.dll.so: cannot open shared object file: No such file or directory
/home/lietu/src/cocreators/Example/Example/bin/Debug/net9.0-desktop/runtimes/linux-x64/native/libUSER32.dll.so: cannot open shared object file: No such file or directory
/usr/share/dotnet/shared/Microsoft.NETCore.App/9.0.5/libUSER32.dll.so: cannot open shared object file: No such file or directory
/home/lietu/src/cocreators/Example/Example/bin/Debug/net9.0-desktop/libUSER32.dll.so: cannot open shared object file: No such file or directory
/home/lietu/src/cocreators/Example/Example/bin/Debug/net9.0-desktop/runtimes/linux-x64/native/USER32.dll: cannot open shared object file: No such file or directory
/usr/share/dotnet/shared/Microsoft.NETCore.App/9.0.5/USER32.dll: cannot open shared object file: No such file or directory
/home/lietu/src/cocreators/Example/Example/bin/Debug/net9.0-desktop/USER32.dll: cannot open shared object file: No such file or directory
/home/lietu/src/cocreators/Example/Example/bin/Debug/net9.0-desktop/runtimes/linux-x64/native/libUSER32.dll: cannot open shared object file: No such file or directory
/usr/share/dotnet/shared/Microsoft.NETCore.App/9.0.5/libUSER32.dll: cannot open shared object file: No such file or directory
/home/lietu/src/cocreators/Example/Example/bin/Debug/net9.0-desktop/libUSER32.dll: cannot open shared object file: No such file or directory

   at Windows.Win32.PInvoke.<SetProcessDpiAwarenessContext>g__LocalExternFunction|240_0(DPI_AWARENESS_CONTEXT value)
   at Windows.Win32.PInvoke.<SetProcessDpiAwarenessContext>g__LocalExternFunction|240_0(DPI_AWARENESS_CONTEXT value)
   at Windows.Win32.PInvoke.SetProcessDpiAwarenessContext(DPI_AWARENESS_CONTEXT value)
   at DpiBootstrap.Init()
   at .cctor()
   --- End of inner exception stack trace ---
   at Example.Program.Main(String[] args)
```
- 6.1.0-dev.189 -> SEGV
- 6.1.0-dev.169 -> same USER32.dll from above
- 6.1.0-dev.150 -> OK
- 6.1.0-dev.152 -> OK
- 6.1.0-dev.154 -> USER32.dll
- 6.1.0-dev.161 -> USER32.dll
- 6.1.0-dev.177 -> USER32.dll
- 6.1.0-dev.183 -> SEGV
- 6.1.0-dev.181 -> SEGV
- 6.1.0-dev.179 -> SEGV

This seems to tell me that

- 6.0.146 until 6.1.0-dev.152 were ok
- 6.1.0-dev.154 to 6.1.0-dev.177 had this USER32.dll issue
- 6.1.0-dev.179 and forward segfault, assuming the `FeatureConfiguration.Rendering.UseOpenGLOnX11 = false;` from earlier is a workaround for them

--- a/src/Uno.UI.Runtime.Skia.X11/X11Renderer.cs
+++ b/src/Uno.UI.Runtime.Skia.X11/X11Renderer.cs
@@ -48,6 +48,8 @@ internal abstract class X11Renderer(IXamlRootHost host, X11Window x11Window)
 			return;
 		}
 
+		MakeCurrent();
+
 		if (_lastSize != newSize || _surface is null || _airspaceHelper is null)
 		{
 			_lastSize = newSize;
@@ -56,8 +58,6 @@ internal abstract class X11Renderer(IXamlRootHost host, X11Window x11Window)
 			_airspaceHelper?.Dispose();
 			_airspaceHelper = new X11AirspaceRenderHelper(display, window, width, height);
 		}
-
-		MakeCurrent();
 
 		var canvas = _surface.Canvas;
 

